src/training/train.py

```diff
@@ -48,15 +48,25 @@ def train_one_epoch(model, data, epoch, optimizer, scaler, scheduler, args, tb_w
     device = torch.device(args.device)
     autocast = get_autocast(args.precision)
     cast_dtype = get_cast_dtype(args.precision)
+    is_flava = args.model.startswith('flava')
 
     model.train()
-    loss = ClipLoss(
-        local_loss=args.local_loss,
-        gather_with_grad=args.gather_with_grad,
-        cache_labels=True,
-        rank=args.rank,
-        world_size=args.world_size,
-        use_horovod=args.horovod)
+    if is_flava:
+        loss = FlavaLoss(
+            local_loss=args.local_loss,
```

Comment:
Does it make sense for this loss to be completely distinct? It seems flava is adding on top of clip rather than being distinct

Reply:
yeah, maybe we add all of the losses separately: MIM, MLM, ITM, MMM, GC (would be our standard ClipLoss) and as a parameter you pass in which of those you want to use. Then you could also try things like CLIP + MIM etc. And FLAVA would basically be passing in all of the possible losses

Reply:
I agree! It'd be awesome if we can make it modular and plug-in those losses separately.
Losses like local (token-aware) contrastive loss + MLM are important for biomedical domains too (https://arxiv.org/abs/2204.09817).

Reply:
Agreed. This can be decoupled into multiple losses and then FLAVALoss using those losses as modular plugins. It would be nice to test ablations as well.